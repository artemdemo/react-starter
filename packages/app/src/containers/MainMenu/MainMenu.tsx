import React, { useContext } from 'react';
import classnames from 'classnames';
import { useTranslation } from '../../hooks/useTranslation';
import { AppContext } from '../../contexts/app/AppContext';
import { Navbar } from 'base-ui';
import { NavbarLink } from 'base-ui';

type TProps = {
  className?: string;
};

export const MainMenu: React.FC<TProps> = (props) => {
  const appContext = useContext(AppContext);
  const { t } = useTranslation();

  const handleVersionClick = () => {
    console.log(ENV);
  };

  return (
    <Navbar
      className={classnames(props.className, 'mb-3')}
      sideContent={
        <NavbarLink
          className="text-muted"
          onClick={handleVersionClick}
          data-testid="mainMenu-app-version"
        >
          {appContext.appVersion}
        </NavbarLink>
      }
      data-testid="mainMenu"
    >
      {/* I need here `div` in order to support flex styling from the `Navbar` */}
      <NavbarLink to="/" data-testid="mainMenu-main">
        {t('main')}
      </NavbarLink>
      <NavbarLink to="/campaigns" data-testid="mainMenu-campaigns">
        {t('campaigns')}
      </NavbarLink>
      <NavbarLink to="/components" data-testid="mainMenu-components">
        {t('components')}
      </NavbarLink>
    </Navbar>
  );
};
