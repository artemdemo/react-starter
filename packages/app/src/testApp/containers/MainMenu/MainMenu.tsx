import React, { useContext } from 'react';
import classnames from 'classnames';
import { useTranslation } from '../../../hooks/useTranslation';
import { AppContext } from '../../../contexts/app/AppContext';
import { Navbar } from 'base-ui';
import { NavbarLink } from 'base-ui';

type TProps = {
  className?: string;
};

export const MainMenu: React.FC<TProps> = (props) => {
  const appContext = useContext(AppContext);
  const { t } = useTranslation();

  const handleVersionClick = () => {
    console.log(ENV);
  };

  return (
    <Navbar
      className={classnames(props.className, 'mb-3')}
      sideContent={
        <NavbarLink
          className="text-muted"
          onClick={handleVersionClick}
          dataTestid="mainMenu-app-version"
        >
          {appContext.appVersion}
        </NavbarLink>
      }
    >
      {/* I need here `div` in order to support flex styling from the `Navbar` */}
      <NavbarLink to="/" dataTestid="mainMenu-main">
        {t('main')}
      </NavbarLink>
      <NavbarLink to="/campaigns" dataTestid="mainMenu-campaigns">
        {t('campaigns')}
      </NavbarLink>
      <NavbarLink to="/components" dataTestid="mainMenu-components">
        {t('components')}
      </NavbarLink>
    </Navbar>
  );
};
