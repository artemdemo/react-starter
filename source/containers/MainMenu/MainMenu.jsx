import React from 'react';
import { Link } from 'react-router-dom';

const MainMenu = () => {
    return (
        <ul className='mt-3'>
            <li><Link to='/'>Main page</Link></li>
            <li><Link to='/json'>JSON test page</Link></li>
            <li><Link to='/campaigns'>Campaigns page</Link></li>
        </ul>
    );
};

export default MainMenu;
