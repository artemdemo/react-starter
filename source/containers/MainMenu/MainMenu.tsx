import React from 'react';
import classnames from 'classnames';
import Navbar from '../../components/Navbar/Navbar';
import NavbarLink from '../../components/Navbar/NavbarLink';
import { t } from '../../services/i18n';

type TProps = {
    className: string;
};

type TState = {};

class MainMenu extends React.PureComponent<TProps, TState> {
    static defaultProps = {
        className: undefined,
    };

    handleVersionClick = () => {
        console.log(ENV);
    };

    render(): React.ReactNode {
        return (
            <Navbar className={classnames(this.props.className, 'mb-3')}>
                {/* I need here `div` in order to support flex styling from the `Navbar` */}
                <div>
                    <NavbarLink to='/'>{t('main')}</NavbarLink>
                    <NavbarLink to='/campaigns'>{t('campaigns')}</NavbarLink>
                    <NavbarLink to='/components'>{t('components')}</NavbarLink>
                </div>
                <div>
                    <NavbarLink onClick={this.handleVersionClick}>{ENV.version}</NavbarLink>
                </div>
            </Navbar>
        );
    }
}

export default MainMenu;
