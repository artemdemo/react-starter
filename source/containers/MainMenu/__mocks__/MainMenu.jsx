import React from 'react';

const MainMenu = () => (
    <div data-mock='MainMenu' />
);

export default MainMenu;
