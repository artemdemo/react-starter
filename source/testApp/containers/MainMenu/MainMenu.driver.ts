import { RenderResults } from '../../../__tests__/testRender';
import { testIdExistsAsync } from '../../../__tests__/utils';

export class MainMenuDriver {
  constructor(public readonly component: RenderResults) {}

  async getAppVersion(): Promise<string | null> {
    const { findByTestId } = this.component;
    const el = await findByTestId('mainMenu-app-version');
    return el.textContent;
  }

  private async testIdExists(key: string): Promise<boolean> {
    return testIdExistsAsync(this.component, key);
  }

  async mainBtnExists(): Promise<boolean> {
    return this.testIdExists('mainMenu-main');
  }

  async campaignsBtnExists(): Promise<boolean> {
    return this.testIdExists('mainMenu-campaigns');
  }

  async componentsBtnExists(): Promise<boolean> {
    return this.testIdExists('mainMenu-components');
  }
}
