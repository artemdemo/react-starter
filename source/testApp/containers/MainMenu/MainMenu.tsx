import React, { useContext } from 'react';
import classnames from 'classnames';
import { Navbar } from '../../../components/Navbar/Navbar';
import { NavbarLink } from '../../../components/Navbar/NavbarLink';
import { t } from '../../../services/i18n';
import { AppContext } from '../../contexts/AppContext';

type TProps = {
  className?: string;
};

export const MainMenu: React.FC<TProps> = (props) => {
  const appContext = useContext(AppContext);

  const handleVersionClick = () => {
    console.log(ENV);
  };

  return (
    <Navbar
      className={classnames(props.className, 'mb-3')}
      sideContent={
        <NavbarLink
          className="text-muted"
          onClick={handleVersionClick}
          dataTestid="mainMenu-app-version"
        >
          {appContext.appVersion}
        </NavbarLink>
      }
    >
      {/* I need here `div` in order to support flex styling from the `Navbar` */}
      <NavbarLink to="/" dataTestid="mainMenu-main">
        {t('main')}
      </NavbarLink>
      <NavbarLink to="/campaigns" dataTestid="mainMenu-campaigns">
        {t('campaigns')}
      </NavbarLink>
      <NavbarLink to="/components" dataTestid="mainMenu-components">
        {t('components')}
      </NavbarLink>
    </Navbar>
  );
};
