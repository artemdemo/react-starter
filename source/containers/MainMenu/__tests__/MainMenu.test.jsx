import React from 'react';
import renderer from 'react-test-renderer';
import MainMenu from '../MainMenu';

/*
 * <MemoryRouter />
 * A <Router> that keeps the history of your “URL” in memory (does not read or write to the address bar).
 * Useful in tests and non-browser environments like React Native.
 *
 * @link https://reacttraining.com/react-router/web/api/MemoryRouter
 */

describe('<MainMenu>', () => {
    it('Render', () => {
        const tree = renderer.create(
            <MainMenu />,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
