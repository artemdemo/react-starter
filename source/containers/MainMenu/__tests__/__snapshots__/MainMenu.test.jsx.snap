// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainMenu> Render 1`] = `
<div
  data-mock="Navbar"
>
  {
  "className": "mb-3"
}
  <div>
    <div
      data-mock="NavbarLink"
    >
      {
  "to": "/"
}
      Main
    </div>
    <div
      data-mock="NavbarLink"
    >
      {
  "to": "/campaigns"
}
      Campaigns
    </div>
    <div
      data-mock="NavbarLink"
    >
      {
  "to": "/components"
}
      Components
    </div>
  </div>
  <div>
    <div
      data-mock="NavbarLink"
    >
      {}
    </div>
  </div>
</div>
`;
